project/driver_controls/inc/input_event.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// Define common typedefs for the various FSMs to use.
+
+#include "event_queue.h"
+
+// Typedef for the FSM arbitration functions
+typedef bool (*InputEventCheck)(Event *e);
```

Comment:
Might want to pass in a `const Event *e` if you don't plan on changing `e`. Const-correctness is good for code cleanliness.